cmd/configurator/main.go

```diff
@@ -109,40 +106,39 @@ func watchPartitions(ctx context.Context, wg *sync.WaitGroup, slurmClient api.Wo
 			// check which partitions are not yet represented in k8s
 			partitionToCreate := notIn(partitionsResp.Partition, nNames)
 			// creating pods for that partitions
-			if err := createNodeForPartitions(k8sClient, partitionToCreate); err != nil {
+			if err := createNodeForPartitions(k8s, partitionToCreate); err != nil {
 				log.Printf("Can't create partitions  %s", err)
 			}
 
 			// some partitions can be deleted from SLURM, so we need to delete pods
 			// which represent those deleted partitions
 			nodesToDelete := notIn(nNames, partitionsResp.Partition)
-			if err := deleteControllingPod(k8sClient, nodesToDelete); err != nil {
+			if err := deleteControllingPod(k8s, nodesToDelete); err != nil {
 				log.Printf("Can't delete controlling pod %s", err)
 			}
 		}
-
 	}
 }
 
-func createNodeForPartitions(k8sClient *corev1.CoreV1Client, partitions []string) error {
+func createNodeForPartitions(podsGetter corev1.PodsGetter, partitions []string) error {
```

Comment:
any reasons to use pods getter instead client?

Reply:
we have interfacer linter that suggested to use narrower interface here

Reply:
imho, right now it's premature optimization. It doesn't give any benefits in readability. We need to know that k8s client implements pods getter. It would be a good improvement on a bigger code base, but in a scope of a single file, don't think so